PackageExplorer/Controls/MultiSelectTreeView.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Media3D;
+
+namespace PackageExplorer.Controls
+{
+    public class MultiSelectTreeView : TreeView
+    {
+        public MultiSelectTreeView()
+        {
+            GotFocus += OnTreeViewItemGotFocus;
+            PreviewMouseLeftButtonDown += OnTreeViewItemPreviewMouseDown;
+            PreviewMouseLeftButtonUp += OnTreeViewItemPreviewMouseUp;
+        }
+
+        private static TreeViewItem _selectTreeViewItemOnMouseUp;
```

Comment:
We are using the [new nullable reference types feature from C#](https://docs.microsoft.com/dotnet/csharp/nullable-references), so you need to declare this (and a lot of other variables in this file) as such.
If you are using the latest VS release you should see a lot of warnings regarding this.

Also, does it need to be `static`? Maybe you can cast the `sender`s in the events to a `MultiSelectTreeView` and set this instance field?

```suggestion
        private TreeViewItem? _selectTreeViewItemOnMouseUp;
```

Reply:
Gonna try to remake these warnings, but when I tried to change few things, custom component functionality seemed to be a bit broken.